rotors_gazebo_plugins/src/gazebo_fw_dynamics_plugin.cpp

```diff
@@ -0,0 +1,406 @@
+
+  // Apply the calculated forced and moments to the main body link.
+  link_->AddRelativeForce(forces);
+  link_->AddRelativeTorque(moments);
+}
+
+double GazeboFwDynamicsPlugin::NormalizedInputToAngle(
+    const ControlSurface& surface, double input) {
+  return (surface.deflection_max + surface.deflection_min) * 0.5 +
+      (surface.deflection_max - surface.deflection_min) * 0.5 * input;
+}
+
+void GazeboFwDynamicsPlugin::CreatePubsAndSubs() {
+  gzdbg << __PRETTY_FUNCTION__ << " called." << std::endl;
+
+  // Create temporary "ConnectRosToGazeboTopic" publisher and message
+  gazebo::transport::PublisherPtr gz_connect_ros_to_gazebo_topic_pub =
+      node_handle_->Advertise<gz_std_msgs::ConnectRosToGazeboTopic>(
+          "~/" + kConnectRosToGazeboSubtopic, 1);
+  gz_std_msgs::ConnectRosToGazeboTopic connect_ros_to_gazebo_topic_msg;
+
+  // ============================================ //
+  // ==== WIND SPEED MSG SETUP (ROS->GAZEBO) ==== //
+  // ============================================ //
+
+  // Wind speed subscriber (Gazebo).
+  wind_speed_sub_ =
+      node_handle_->Subscribe("~/" + namespace_ + "/" + wind_speed_sub_topic_,
+                              &GazeboFwDynamicsPlugin::WindSpeedCallback,
+                              this);
+
+  connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" +
+                                                wind_speed_sub_topic_);
+  connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" +
+                                                   wind_speed_sub_topic_);
+  connect_ros_to_gazebo_topic_msg.set_msgtype(
+      gz_std_msgs::ConnectRosToGazeboTopic::WIND_SPEED);
+  gz_connect_ros_to_gazebo_topic_pub->Publish(connect_ros_to_gazebo_topic_msg,
+                                              true);
+
+  // If we are using a joystick for control inputs then subscribe to the
+  // RollPitchYawrateThrust msgs, otherwise subscribe to the Actuator msgs.
+  if (is_input_joystick_) {
+    // ========================================================= //
+    // === ROLL_PITCH_YAWRATE_THRUST MSG SETUP (ROS->GAZEBO) === //
+    // ========================================================= //
+    roll_pitch_yawrate_thrust_sub_ =
+        node_handle_->Subscribe("~/" + namespace_ + "/" +
+            roll_pitch_yawrate_thrust_sub_topic_,
+            &GazeboFwDynamicsPlugin::RollPitchYawrateThrustCallback, this);
+
+    connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" +
+        roll_pitch_yawrate_thrust_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" +
+        roll_pitch_yawrate_thrust_sub_topic_);
+    connect_ros_to_gazebo_topic_msg.set_msgtype(
+        gz_std_msgs::ConnectRosToGazeboTopic::ROLL_PITCH_YAWRATE_THRUST);
+  } else {
+    // ============================================ //
+    // ===== ACTUATORS MSG SETUP (ROS->GAZEBO) ==== //
+    // ============================================ //
+
```

Comment:
For many things, especially those involving the Gazebo-ROS interfacing, I followed the formatting of other plugins. So perhaps it's best to consult @gbmhunter regarding this.